### ISSUE TITLE
FNX-3442 ⁃ [Bug] java.lang.IllegalStateException: Migration didn't properly handle top_sites(mozilla.components.feature.top.sites.db.TopSiteEntity)

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/fenix#10334]
I was making local fenix + a-c builds and ran into a migration-related exception on startup. I may have went from a newer build to an older build during the process:

```
    Process: org.mozilla.fennec_aurora, PID: 31059
    java.lang.RuntimeException: Exception while computing database live data.
        at androidx.room.RoomTrackingLiveData$1.run(RoomTrackingLiveData.java:6)
        at androidx.room.TransactionExecutor$1.run(TransactionExecutor.java:1)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.IllegalStateException: Migration didn't properly handle top_sites(mozilla.components.feature.top.sites.db.TopSiteEntity).
     Expected:
    TableInfo{name='top_sites', columns={created_at=Column{name='created_at', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0}, id=Column{name='id', type='INTEGER', affinity='3', notNull=false, primaryKeyPosition=1}, title=Column{name='title', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0}, is_default=Column{name='is_default', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0}, url=Column{name='url', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0}}, foreignKeys=[], indices=[]}
     Found:
    TableInfo{name='top_sites', columns={isDefault=Column{name='isDefault', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0}, created_at=Column{name='created_at', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0}, id=Column{name='id', type='INTEGER', affinity='3', notNull=false, primaryKeyPosition=1}, title=Column{name='title', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0}, url=Column{name='url', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0}}, foreignKeys=[], indices=[]}
        at mozilla.components.feature.top.sites.db.TopSiteDatabase_Impl$1.validateMigration(TopSiteDatabase_Impl.java:12)
        at androidx.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.java:6)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.java:4)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:417)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:317)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:2)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:1)
        at androidx.room.RoomDatabase.beginTransaction(RoomDatabase.java:2)
        at mozilla.components.feature.top.sites.db.TopSiteDao_Impl$3.call(TopSiteDao_Impl.java:3)
        at androidx.room.RoomTrackingLiveData$1.run(RoomTrackingLiveData.java:5)
        	... 4 more
```

This could have been caused by my local changes but I wanted to report it just in case it wasn't. CC @pocmo 

* Android device: Pixel 2
* Fenix version: 4006501d4103fce67cefc0cb25bb60b6492dc1db (Apr 28) + 1d04bb79acc41890ba42e4b47d2fc5ab67371d9a in a-c (Apr 29)

**Comment by TitanNano** [comment]
I'm also getting this after installing the latest nightly build.

**Comment by eliserichards** [comment]
Should be fixed in #10336
